rpcs3/rpcs3qt/emu_settings.cpp

```diff
@@ -307,6 +307,49 @@ void emu_settings::EnhanceCheckBox(QCheckBox* checkbox, emu_settings_type type)
 	});
 }
 
+void emu_settings::EnhanceDateTimeEdit(QDateTimeEdit* date_time_edit, emu_settings_type type, const QString& format, bool calendarMode)
+{
+	if (!date_time_edit)
+	{
+		cfg_log.fatal("EnhanceDateTimeEdit '%s' was used with an invalid object", cfg_adapter::get_setting_name(type));
+		return;
+	}
+
+	date_time_edit->setDisplayFormat(format);
+	date_time_edit->setCalendarPopup(calendarMode);
+
+	bool ok_def = false;
+	const s64 def = qstr(GetSettingDefault(type)).toLongLong(&ok_def);
+	if (!ok_def)
+	{
+		cfg_log.fatal("EnhanceDateTimeEdit '%s' was used with an invalid emu_settings_type", cfg_adapter::get_setting_name(type));
+		return;
+	}
+
+	bool ok_sel = false;
+	s64 val = qstr(GetSetting(type)).toLongLong(&ok_sel);
+
+	if (!ok_sel)
+	{
+		cfg_log.error("EnhanceDateTimeEdit '%s' tried to set an invalid value: %d. Setting to default: %d", cfg_adapter::get_setting_name(type), val, def);
+		val = def;
+		m_broken_types.insert(type);
+		SetSetting(type, sstr(def));
+	}
+
+	// PS3 RTC ticks are in microseconds, so we'll store our times in this format, but QT likes milliseconds, so we throw out the little remainder
```

Comment:
I don't think this function has any context. So talking about a specific use case here seems odd.
It also seems odd to use this addition when this function is only supposed to get and set a time.
Maybe add an offset param instead

Reply:
Comment was out of date also.  Have generalised function so that it can behave either as 'offset from current time' as here, or just as a direct date/time mode.  I'm using the same callback for both, just to keep the logic a little less confusing (I think having two different callbacks only one of which is conditionally connected would be confusing).

Reply:
Darn... it looks like QDateTime string formatting wants another Qt library linked in...
I added the QDateTime formatting to make sense of having an argument called as_offset_from_now being false..
i.e. it would load/save settings in terms of fixed date/times rather than the seconds offset that I have right now [this was point e. in my known issues]